feat(api): add user account activation route

meardagoat opened the pull request:
## feat(api): add user account activation route

This PR implements the user account activation route as described in issue #95.

### Purpose
Allow users to activate their account via an email link containing a JWT token.

### Changes

**Backend:**
- Added `findUserById()` repository method to retrieve a user by ID
- Created `activateUserController` to verify token and activate user
- Added route `GET /api/authentication/activate?token=...`

**Activation logic:**
- Returns `201` when activation succeeds
- Returns `400` when token is missing or invalid
- Returns `401` when user does not exist or is already active

**Tests:**
- Integration tests for `findUserById()`
- Acceptance tests covering all activation route scenarios

Closes #95

theotime2005 reviewed (changes requested):
It's an excellent pull request, we feel that the increase in skills is there and the reading of the rest of the code base too. However, pay attention to the following points:
- Take back the comments and apply them, or discuss them in the discussion threads to keep a history,
- Separate the commits: There is only one commit for the entire PR, it would take a commit for the repository and its test, and the same for the controller and the route.
It's VERY GOOD! but I can approve before fixing.

meardagoat committed:
feat(api): add repository methods for user activation

Add findUserById() and activateUserById() methods to user repository.

- findUserById(): retrieves a user by their ID, returns null if not found
- activateUserById(): sets isActive to true for a user, throws error if user not found

Integration tests cover both methods with success and error cases.

meardagoat committed:
feat(api): add user account activation route

Implement user account activation via email token.

Controller:
- Create activateUserController with JWT token verification
- Validate user existence and activation status
- Return appropriate status codes (201, 400, 401, 500)

Routes:
- Add GET /api/authentication/activate?token=... endpoint
- Integrate activation route into authentication routes

Tests:
- Unit tests for controller covering all scenarios
- Acceptance tests for the complete activation flow

Closes #95

theotime2005 reviewed (changes requested):
Try again!

meardagoat commented:
@theotime2005 i did the 

```git pull --rebase```

theotime2005 commented on api/src/identities-access-management/controllers/activate-user-controller.js:
```diff
@@ -0,0 +1,67 @@
+      logger.info("Activation attempt for already active user", { userId });
+      return res.status(401).json({ error: ERRORS.USER.NOT_FOUND_OR_ALREADY_ACTIVE });
+    }
+
```
Ok for me because repository do not logged an error

theotime2005 commented on api/src/identities-access-management/controllers/activate-user-controller.js:
```diff
@@ -0,0 +1,67 @@
+      return res.status(401).json({ error: ERRORS.USER.NOT_FOUND_OR_ALREADY_ACTIVE });
+    }
+
+    // Activate the user
```
Remove this comment

theotime2005 reviewed (changes requested):
It's so good, but you can:
- Apply suggestions,
- Rewrite your commit history.

theotime2005 commented:
> It's so good, but you can:
> 
> * Apply suggestions,
> * Rewrite your commit history.

@meardagoat let's go!

theotime2005 reviewed (approved):
It's reviewed! merge it :rocket:

edp5-service commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/edp5/kompagnon/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
